### PR TITLE
Document invocant in method signature

### LINKED ISSUE #1121: Document use of method invocant to distinguish between class and object methods
See also perl6/roast [#217](https://github.com/perl6/roast/issues/217) and [IRC](https://irclog.perlgeek.de/perl6/2017-01-05#i_13862614).

The technique is sometimes used by examples in the docs but not really described in either of [Classes and Objects](https://docs.perl6.org/language/classtut) or [Object Orientation](https://docs.perl6.org/language/objects).  It would seem helpful to have it discussed directly someplace where someone might look for it.

```Perl6
class C {
    multi method f(::?CLASS:U:){say "class method"}
    multi method f(::?CLASS:D:){say "object method"}
}
C.f;       # says class method
C.new.f;   # says object method
```

### PR DESCRIPTION
**Pull request by zakame**
Add documentation on setting invocant in method signatures, using type
constraints to define class and object methods, and using multi
declarator for defining same-name class and object methods.

Fixes #1121.

### CODE REVIEW

**Comment by zakame** [comment]
Took a bit longer than I expected, as I realized it needs documenting the concept of invocant first, and a slight peek at type constraints, to better illustrate the entire concept.

Let me know if it needs more tuning, or more :pizza:

File: doc/Language/objects.pod6 [hunk 1]
```diff
@@ -264,6 +264,54 @@ Method names can be resolved at runtime with the C<.""> operator.
     A.new."$name"().say;
     # OUTPUT: «(Any)␤»
 
+A method's signature can have an I<invocant> as its first parameter
+followed by a comma, which allows for the method to refer to the object
```
Comment (moritz) [inline doc/Language/objects.pod6]: This should be "colon", not "comma".

Comment (zakame) [reply doc/Language/objects.pod6]: Doh! 🤦‍♂️ Thanks for the catch @moritz !

**Comment by zakame** [comment]
Thanks again @moritz !

**Comment by moritz** [comment]
Thank you for your contribution @zakame! You should have received an invitation to the "perl6" team, which allows you to push to the doc repo (and others) directly.

**Comment by zakame** [comment]
Yep, just saw it, many thanks! 😸

**Comment by ronaldxs** [comment]
Suggestion: the new patch should be in a separate section titled/labeled "Class and Instance methods" after the "method" and "self" sections.  I think having a new section better fits the structure of the surrounding pieces of the OO document.

**Comment by zakame** [comment]
@ronaldxs yeah I was mulling that too, thanks for suggesting 👍

**Pull request merged**